cmd/preview/wake.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package preview
+
+import (
+	"context"
+	"fmt"
+
+	contextCMD "github.com/okteto/okteto/cmd/context"
+	"github.com/okteto/okteto/cmd/utils"
+	oktetoErrors "github.com/okteto/okteto/pkg/errors"
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/spf13/cobra"
+)
+
+func Wake(ctx context.Context) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "wake <name>",
+		Short: "Wakes a preview",
+		Args:  utils.ExactArgsAccepted(1, ""),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			prToWake := args[0]
+			if err := contextCMD.NewContextCommand().Run(ctx, &contextCMD.ContextOptions{}); err != nil {
+				return err
+			}
+
+			if !okteto.IsOkteto() {
+				return oktetoErrors.ErrContextIsNotOktetoCluster
+			}
+
+			prCmd, err := NewCommand()
+			if err != nil {
+				return err
+			}
+			err = prCmd.ExecuteWakePreview(ctx, prToWake)
+			return err
+		},
+	}
+	return cmd
+}
+
+func (pr *Command) ExecuteWakePreview(ctx context.Context, preview string) error {
+	// Spinner to be loaded before waking a preview
+	oktetoLog.Spinner(fmt.Sprintf("Waking %s preview", preview))
+	oktetoLog.StartSpinner()
+	defer oktetoLog.StopSpinner()
+
+	// trigger preview to wake
+	if err := pr.okClient.Previews().WakePreview(ctx, preview); err != nil {
```

Comment:
The endpoint is the same as Namespace so it would be:
```suggestion
	if err := pr.okClient.Namespaces().Wake(ctx, preview); err != nil {
```

Reply:
Hey @jLopezbarb! Sorry I didn't understand, why we need a `Wake` endpoint over here. The wake endpoint requires a `namespaceClient` whereas for the preview command, don't we need to wake a `previewClient`?
If we want to use a `Wake` then are we trying to wake a namespace over here?
Sorry, I am new to this project and just have some doubts.

Reply:
Thank you for clearing up my doubt!